driver/vm.go

```diff
@@ -546,27 +546,27 @@ func addDisks(_ *Driver, devices object.VirtualDeviceList, config *CreateConfig)
 		}
 
 		if dc.DiskType == "thin" {
-			dc.DiskEagerlyScrub    = false
-			dc.DiskThinProvisioned = true
+			dc.diskEagerlyScrub    = false
+			dc.diskThinProvisioned = true
 		} else if dc.DiskType == "thick_eager" {
-			dc.DiskEagerlyScrub    = true
-			dc.DiskThinProvisioned = false
+			dc.diskEagerlyScrub    = true
+			dc.diskThinProvisioned = false
 		} else if dc.DiskType == "thick_lazy" {
-			dc.DiskEagerlyScrub    = false
-			dc.DiskThinProvisioned = false
+			dc.diskEagerlyScrub    = false
+			dc.diskThinProvisioned = false
 		} else {
 			// default disk type: Thick provisioned lazy zeroed
-			dc.DiskEagerlyScrub    = false
-			dc.DiskThinProvisioned = false
+			dc.diskEagerlyScrub    = false
```

Comment:
Let's merge that branch with previous `thick_lazy`. Also I'd recommend to move whole if-else to `CreateConfig#Prepare` which is called from 'prepare' state of step

Reply:
The reason I wrote it like this, is to have a default value for the disk type and also explicitly naming all the possible options. Also, if the default is ever changed, the values for thick_lazy are already there. But I agree it is not strictly necessary to keep it like this.

I don't know if this conditional can be moved elsewhere, I would have to look into that. The reason it is here is to be able to specify a disk type per individual disk.

Reply:
The disk type logic has been moved to function CreateConfig